File: lib/LH.java
package krpc.client.services;

import com.google.protobuf.ByteString;

import krpc.client.Connection;
import krpc.client.Encoder;
import krpc.client.RemoteEnum;
import krpc.client.RemoteObject;
import krpc.client.RPCInfo;
import krpc.client.RPCException;
import krpc.client.Types;

public class LH {

    public static LH newInstance(Connection connection) {
        return new LH(connection);
    }

    private Connection connection;

    private LH(Connection connection) {
        this.connection = connection;
    }

    private void addExceptionTypes(Connection connection) {
    }

    @SuppressWarnings({ "unchecked" })
    @RPCInfo(service = "LH", procedure = "Height", types = _Types.class)
    public double height() throws RPCException {
        ByteString _data = this.connection.invoke("LH", "Height");
        return (double) Encoder.decode(_data, krpc.client.Types.createValue(krpc.schema.KRPC.Type.TypeCode.DOUBLE), this.connection);
    }

    public static class _Types {
        public static krpc.schema.KRPC.Type getReturnType(String procedure) {
            switch (procedure) {
            case "Height":
                return krpc.client.Types.createValue(krpc.schema.KRPC.Type.TypeCode.DOUBLE);
            }
            throw new IllegalArgumentException("Procedure '" + procedure +"' not found");
        }

        public static krpc.schema.KRPC.Type[] getParameterTypes(String procedure) {
            switch (procedure) {
            case "Height":
                return new krpc.schema.KRPC.Type[] {
                };
            }
            throw new IllegalArgumentException("Procedure '" + procedure +"' not found");
        }
    }
}

File: lib/LH.cs
using Google.Protobuf;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;

#if NET35
using systemAlias = global::KRPC.Client.Compatibility;
using genericCollectionsAlias = global::KRPC.Client.Compatibility;
#else
using systemAlias = global::System;
using genericCollectionsAlias = global::System.Collections.Generic;
#endif

namespace KRPC.Client.Services.LH
{
    /// <summary>
    /// Extension methods for LH service.
    /// </summary>
    public static class ExtensionMethods
    {
        /// <summary>
        /// Create an instance of the LH service.
        /// </summary>
        public static global::KRPC.Client.Services.LH.Service LH (this global::KRPC.Client.IConnection connection)
        {
            return new global::KRPC.Client.Services.LH.Service (connection);
        }
    }

    /// <summary>
    /// LH service.
    /// </summary>
    public class Service
    {
        global::KRPC.Client.IConnection connection;

        internal Service (global::KRPC.Client.IConnection serverConnection)
        {
            connection = serverConnection;
        }

        [global::KRPC.Client.Attributes.RPCAttribute ("LH", "Height")]
        public double Height ()
        {
            ByteString _data = connection.Invoke ("LH", "Height");
            return (double)global::KRPC.Client.Encoder.Decode (_data, typeof(double), connection);
        }
    }
}

